connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala

```diff
@@ -79,7 +107,11 @@ private[sql] class SparkConnectClient(
       .setSessionId(sessionId)
       .setClientType(userAgent)
       .build()
-    stub.executePlan(request)
+    retry {
+      val result = stub.executePlan(request)
+      result.hasNext // moves evaluation of BlockingResponseStream to SparkConnectClient
```

Comment:
could you elaborate on why this is needed? I believe this `.hasNext` can block for quite a while until the first response comes back on the stream.

Reply:
The problem was that `execute` returns a lazily evaluated `java.util.Iterator[proto.ExecutePlanResponse]` that is then used in `SparkResult.processResponses` where `responses.hasNext` fails with a `StatusRuntimeException` outside of  the `retry` block. As suggested by Hyukjin, I now created a custom Iterator that solves this problem.